internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,257 @@
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
+		}
+		return envDescriber, nil
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application %s: %w", o.appName, err)
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	//TODO(varun359): Get build information from the manifest and task definition for workloads
+
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		return o.validateEnvName()
+	}
+	envs, err := o.getDeployedEnvironments()
+	switch {
+	case err != nil:
+	case len(envs) == 0:
+	case len(envs) == 1:
+	}
+	switch {
+	case err != nil:
+		return err
+	case len(envs) == 0:
+		return fmt.Errorf("no deployed environments found in the app %s", o.appName)
+	case len(envs) == 1:
+		log.Infof("Only one environment found, defaulting to: %s\n", color.HighlightUserInput(envs[0]))
+		o.envName = envs[0]
+		return nil
+
+	}
+	selectedEnvName, err := o.prompt.SelectOne("Select an environment in which you want to test", "", envs, prompt.WithFinalMessage("Environment:"))
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = selectedEnvName
+	return nil
+}
+
+func (o *localRunOpts) getDeployedEnvironments() ([]string, error) {
+	envConfig, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("get environments for app %s: %w", o.appName, err)
+	}
+
+	var deployedEnvs []string
+	for _, env := range envConfig {
+		isDeployed, err := o.isEnvironmentDeployed(env.Name)
+		if err != nil {
+			return nil, err
+		}
+		if isDeployed {
+			deployedEnvs = append(deployedEnvs, env.Name)
+		}
+	}
+
+	return deployedEnvs, nil
+}
+
+func (o *localRunOpts) isEnvironmentDeployed(envName string) (bool, error) {
+	var checker versionCompatibilityChecker
+
+	checker, err := o.envVersionChecker(envName)
+	if err != nil {
+		return false, err
+	}
+
+	currVersion, err := checker.Version()
+	if err != nil {
+		return false, fmt.Errorf("get environment %q version: %w", envName, err)
+	}
+	if currVersion == version.EnvTemplateBootstrap {
+		return false, nil
+	}
+	return true, nil
+}
+func (o *localRunOpts) validateEnvName() error {
+	isDeployed, err := o.isEnvironmentDeployed(o.envName)
+	if err != nil {
+		return err
+	}
+	if !isDeployed {
+		return fmt.Errorf(`cannot use an environment which is not deployed Please run "copilot env deploy, --name %s" to deploy the environment first`, o.envName)
+	}
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskWorkloadName() error {
```

Comment:
qq: I know we require the workload to be deployed - does the workload have to be in the current workspace? Same question for environment.

Reply:
> what if the workload got deleted locally

Then it won't appear in the value returned by `o.ws.ListWorkloads`, and we won't think that the workload is "in the workspace"